src/test/java/org/jabref/logic/importer/fetcher/CompositeSearchBasedFetcherTest.java

```diff
@@ -0,0 +1,110 @@
+
+        List<BibEntry> queryResult = compositeFetcher.performSearch("");
+
+        Assertions.assertTrue(queryResult.isEmpty());
+    }
+
+    @ParameterizedTest(name = "Perform search on query \"quantum\". Using the CompositeFetcher of the following " +
+            "Fetchers: {arguments}")
+    @MethodSource("performSearchParameters")
+    public void performSearchOnNonEmptyQuery(Set<SearchBasedFetcher> fetchers) {
+        CompositeSearchBasedFetcher compositeFetcher = new CompositeSearchBasedFetcher(fetchers);
+
+        List<BibEntry> compositeResult = compositeFetcher.performSearch("quantum");
+
+        for (SearchBasedFetcher fetcher : fetchers) {
+            try {
+                Assertions.assertTrue(compositeResult.containsAll(fetcher.performSearch("quantum")));
+            } catch (FetcherException e) {
+                /* We catch the Fetcher exception here, since the failing fetcher also fails in the CompositeFetcher
+                 * and just leads to no additional results in the returned list. Therefor the test should not fail
+                 * due to the fetcher exception
+                 */
+                LOGGER.warn(String.format("Fetcher %s failed ", fetcher.getName()), e);
+            }
+        }
+    }
+
+    /**
+     * This method provides other methods with different sized sets of Searchbased fetchers wrapped in Arguments.
+     *
+     * @return A stream of Arguments wrapping the sets.
+     */
+    static Stream<Arguments> performSearchParameters() {
+        ImportFormatPreferences importFormatPreferences = mock(ImportFormatPreferences.class);
+        when(importFormatPreferences.getFieldContentFormatterPreferences())
+                .thenReturn(mock(FieldContentFormatterPreferences.class));
+        List<Set<SearchBasedFetcher>> fetcherParameters = new ArrayList<>();
+        List<SearchBasedFetcher> list = new ArrayList<>();
+
+        list.add(new ArXiv(importFormatPreferences));
+        list.add(new INSPIREFetcher(importFormatPreferences));
+        list.add(new GvkFetcher());
+        list.add(new AstrophysicsDataSystem(importFormatPreferences));
+        list.add(new MathSciNet(importFormatPreferences));
+        list.add(new ZbMATH(importFormatPreferences));
+        list.add(new GoogleScholar(importFormatPreferences));
+        list.add(new DBLPFetcher(importFormatPreferences));
+        list.add(new SpringerFetcher());
+        list.add(new CrossRef());
+        list.add(new CiteSeer());
+        list.add(new DOAJFetcher(importFormatPreferences));
+        list.add(new IEEE(importFormatPreferences));
+        /* Disabled due to issue regarding Comparison: Title fields of the entries that otherwise are equivalent differ
+         * due to different JAXBElements.
+         */
+        // list.add(new MedlineFetcher());
+
+        // Create different sized sets of fetchers to use in the composite fetcher.
+        for (int i = 1; i < list.size(); i++) {
```

Comment:
Can we do some different sub sets? Not always include the first one, but something like: Always +3 and then use the subsets matched by the respective bits?

```
6543210
0110001 --> list entry 5,4,0 are used
```

(Not sure what the others think of it)